src/SingleEventManager.lua

```diff
@@ -53,7 +53,17 @@ function SingleEventManager:connectEvent(key, listener)
 end
 
 function SingleEventManager:connectPropertyChange(key, listener)
-	local event = self._instance:GetPropertyChangedSignal(key)
+	local success, event = pcall(function()
+		return self._instance:GetPropertyChangedSignal(key)
+	end)
+
+	if not success then
+		error(("Cannot subscribe to changed signal: %q is not a property of %s"):format(
```

Comment:
This isn't necessarily what the actual error is, since it could be a permissions problem. :(

Do we want to do some sort of really nasty string matching to try to fix those problems, or are we okay with that edge case?

Reply:
For the permissions problem, do you mean that `GetPropertyChangedSignal` could throw if you ask for a specific property that is guarded by a certain level? Because the function itself isn't behind any level right? If it's true then I am less concerned about it because it's almost true that the property does not exist, in that permission context let's say.

One thing we could do is print out the original error, like appending `(original error thrown: %s)` to the message and pass the returned error message.

Or we could simply append to the message `(or lacking permission level)` too.

And I agree that the string matching on an error message might be too fragile (and also not elegant)